src/apps/DEL_DOT_VEC_2D-Sycl.cpp

```diff
@@ -0,0 +1,88 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-24, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "DEL_DOT_VEC_2D.hpp"
+
+#include "RAJA/RAJA.hpp"
+
+#if defined(RAJA_ENABLE_SYCL)
+
+#include "AppsData.hpp"
+
+#include <iostream>
+
+#include "common/SyclDataUtils.hpp"
+
+namespace rajaperf 
+{
+namespace apps
+{
+
+template <size_t work_group_size >
+void DEL_DOT_VEC_2D::runSyclVariantImpl(VariantID vid)
+{
+  const Index_type run_reps = getRunReps();
+  const Index_type ibegin = 0;
+  const Index_type iend = m_domain->n_real_zones;
+
+  auto res{getSyclResource()};
+
+  DEL_DOT_VEC_2D_DATA_SETUP;
+
+  if ( vid == Base_SYCL ) {
+
+    startTimer();
+    for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
+
+      const size_t grid_size = work_group_size * RAJA_DIVIDE_CEILING_INT(iend, work_group_size);
+
+      qu->submit([&] (sycl::handler& h) {
+        h.parallel_for(sycl::nd_range<1> (grid_size, work_group_size),
+                       [=] (sycl::nd_item<1> item) {
+
+          Index_type ii = item.get_global_id(0);
+          if (ii < iend) {
+            DEL_DOT_VEC_2D_BODY_INDEX
+            DEL_DOT_VEC_2D_BODY
+          }
+
+        });
+      });
+
+    }
+    qu->wait();
+    stopTimer();
```

Comment:
hmm, this is probably fine, but normally we hide the wait in `stopTimer`.

Reply:
Will be removed in a subsequent PR when https://github.com/LLNL/RAJAPerf/issues/434 is resolved